### PR TITLE
Ci to ci2 merge

### PR DESCRIPTION
**Pull request by rbuckland**
This merge brings 3 significant changes

- all .bsh files are moved into a src/ folder. This meant that some of the logic of tests had to change.
- the tests can execute locally on OSX (and on Linux)
  - this was a multi pronged approach whereby some commands (grep, sed, readlink) had to be executed with "like/for/like" semantics for OSX and Linux. Mostly changes in either tests of lib-ci we added for this
- enhanced the tests to do "temp dirs" and "temp files" consistently using a common "Mktemp_Portable" function in lib-ci. This then enabled a 
   - cleanup routine on test execution.
  - enhancement to - any testing "git" type work is performed in an "out of" project folder (eg /tmp/somepath/fake-repo)

### CODE REVIEW

File: src/docker-support.bsh [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+import lib-ci
+
+CI_Env_Adapt $(CI_Env_Get)
+
+function usage() {
+  echo "$0 setup | publish <options>" 1>&2
+  echo 1>&2
+  echo "The two simplest forms are '$0 setup' or '$0 publish'. All other vars are searched on the environment" 1>&2
+  echo 1>&2
```
Comment (stephanh) [inline src/docker-support.bsh]: Can you add an example block for setup and publish with the environment variables defined.

File: src/setup-version.bsh [hunk 2]
```diff
@@ -28,16 +28,6 @@ function import() {
...
-    exit 1
-fi
-
-new_version=$(Version_Get $(cat VERSION))
-if [ -z $new_version ]; then
-    exit 1
-fi
-echo "Version Mapped: $version => $new_version"
-echo "$new_version" > VERSION
+Version_Write_New
```
Comment (stephanh) [inline src/setup-version.bsh]: Just curious why did you move that to the library?

File: src/docker-support.bsh (hunk 1 above)
Comment (rbuckland) [inline src/docker-support.bsh]: Yes - I will write a README.

**Comment by stephanh** [comment]
Nicely done :+1:
